packages/aggregator/src/analytics/aggregateAnalytics/adjustOptionsToAggregationList.js

```diff
@@ -83,26 +79,34 @@ const getAdjustedOrganisationUnitsAndAggregations = async (
       } = aggregation.config;
       const [
         dataSourceEntities,
-        relations,
-      ] = await entityConnection.getDataSourceEntitiesAndRelations(
+        relationships,
+      ] = await entityConnection.getDataSourceEntitiesAndRelationships(
         hierarchy,
         organisationUnitCodes,
         aggregationEntityType,
         dataSourceEntityType,
         dataSourceEntityFilter,
       );
 
-      if (!adjustedOrganisationUnitCodes) {
-        adjustedOrganisationUnitCodes = dataSourceEntities;
-      }
-      adjustedAggregations.push({
-        ...aggregation,
-        config: { ...aggregation.config, orgUnitMap: relations },
-      });
+      return [
+        {
+          ...aggregation,
+          config: { ...aggregation.config, orgUnitMap: relationships },
+        },
+        dataSourceEntities,
+      ];
     }),
   );
 
-  return [adjustedOrganisationUnitCodes, adjustedAggregations];
+  const adjustedAggregations = adjustedAggregationsAndDataSourceEntites.map(
+    aggregationAndDataSourceEntities => aggregationAndDataSourceEntities[0],
+  );
+  const dataSourceEntities =
+    adjustedAggregationsAndDataSourceEntites
+      .map(aggregationAndDataSourceEntities => aggregationAndDataSourceEntities[1])
+      .find(dataSourceEntities => dataSourceEntities !== undefined) || organisationUnitCodes;
```

Comment:
Just checking, shouldn't this be:
```suggestion
      .filter(dataSourceEntities => dataSourceEntities !== undefined).flat() || organisationUnitCodes;
```

Also `dataSourceEntities` is already declared the scope above

Reply:
So the logic to determine the dataSourceEntities for an array of aggregations is actually a little bit wonky at the moment... Basically what we do is find the first entity aggregation, get its dataSourceEntities, and use that. In a sense, dataSourceEntities is really just a fetchOption, and its a miscategorisation that we have to put them in an aggregation just to get the fetch working correctly. I've already had to write multiple reports/indicators that have something like:
```
fetch: {
  ...,
  aggregations: [{ type: 'RAW', config: { dataSourceEntityType: 'facility' } }] 
}
```
Just so that I can fetch the data I want. Ideally we'd have something like this instead:
```
fetch: {
  ...,
  organisationUnits: { codes: ['TO', 'PG'], dataSourceEntityType: 'facility' }
}
```

So going back to the logic you raised concerns about, it's actually meant to find the first `not undefined` value (even if that value is an empty array), and ignore all the later ones.

As for the `dataSourceEntities` variable being already declared in scope, I think it's okay as those declarations are in the `Promise.all()` code block?